src/cargo/core/compiler/mod.rs

```diff
@@ -1493,6 +1495,25 @@ fn package_remap(build_runner: &BuildRunner<'_, '_>, unit: &Unit) -> OsString {
     remap
 }
 
+/// Remap all paths pointing to `build.build-dir`,
+/// i.e., `[BUILD_DIR]/debug/deps/foo-[HASH].dwo` would be remapped to
+/// `debug/deps/foo-[HASH].dwo`.
```

Comment:
> The open question is what we want to remap to, to help debugger to find the source files.

If we could assume things were in `target` then we could just add that as a prefix and make this like the pkg root remapping.  People can match on that and remap it in their debugger.

As we move the build dir, this will be less likely to be a match.  This makes me wonder if we should be forward looking and name it something more contextually meaningful like `build-dir`.

Reply:
What about a `/cargo/build-dir` prefix. We don't need to make the final call right now. This eventually needs to re-review before stabilization.

We might probably want to do that for other kinds of dependencies.

Reply:
Whatever seems to work best.  I mostly feel we need some kind of prefix